homeassistant/components/emoncms_history.py

```diff
@@ -0,0 +1,92 @@
+"""
+A component which allows you to send data to Emoncms.
+
+For more details about this component, please refer to the documentation at
+https://home-assistant.io/components/emoncms_history/
+"""
+import logging
+from datetime import timedelta
+import json
+
+import voluptuous as vol
+import requests
+
+
+from homeassistant.const import (
+    CONF_API_KEY, CONF_WHITELIST,
+    EVENT_STATE_CHANGED, CONF_URL, STATE_UNKNOWN,
+    STATE_UNAVAILABLE)
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers import state as state_helper
+from homeassistant.util import Throttle
+
+_LOGGER = logging.getLogger(__name__)
+
+DOMAIN = "emoncms_history"
+CONF_INPUTNODE = "inputnode"
+
+MIN_TIME_BETWEEN_UPDATES = timedelta(seconds=1)
+
+CONFIG_SCHEMA = vol.Schema({
+    DOMAIN: vol.Schema({
+        vol.Required(CONF_API_KEY): cv.string,
+        vol.Required(CONF_URL): cv.string,
+        vol.Required(CONF_INPUTNODE): cv.positive_int,
+        vol.Required(CONF_WHITELIST, default=[]):
+            vol.All(cv.ensure_list, [cv.entity_id]),
```

Comment:
i switched to vol.Required(CONF_WHITELIST): cv.entity_ids, which handles everything that's needed the intial changes i did had problem when the user would only write `whitelist:` and no other values giving errors about a nonetype. cv.entity_ids checks for this